src/rsa.rs

```diff
@@ -244,6 +255,49 @@ impl RSAKeyPair {
     pub fn public_modulus_len(&self) -> usize {
         unsafe { RSA_size(self.rsa.as_ref()) }
     }
+
+    /// Sign the given message, hashed with the given digest algorithm.
+    pub fn sign(&self, digest_alg: &'static digest::Algorithm, msg: &[u8], signature: &mut [u8]) -> Result<(), ()> {
+        if signature.len() != self.public_modulus_len() {
+            return Err(());
+        }
+
+        let found = PREFIXES.into_iter().find(|pair| pair.0.nid == digest_alg.nid);
+        if found.is_none() {
+            return Err(());
+        }
+
+        let msg_len = self.public_modulus_len();
+        let hashed = digest::digest(digest_alg, msg);
+        let prefix = found.unwrap().1;
+        let digest_len = prefix.len() + hashed.as_ref().len();
+        if msg_len < digest_len + 11 {
+            return Err(());
+        }
+
+        let padding_len = msg_len - digest_len - 3;
+        let mut pad_bytes: Vec<u8> = Vec::with_capacity(msg_len);
```

Comment:
I think we should do a follow up where we do all of this in-place in `signature`, avoiding the need to allocate this memory at all.